File: Zamba/ZambaFramework/Zamba.FWIL/ISearch.vb
Imports Zamba.Searchs

Public Interface ISearch
    Property Doctypes As Generic.List(Of IDocType)
    Property Indexs As Generic.List(Of IIndex)
    Property CaseSensitive() As Boolean
    Property UseVersion() As Boolean
    Property ShowIndexOnGrid() As Boolean
    Property Textsearch() As String
    Property ParentName() As String
    Property Name() As String
    Property MaxResults() As Int32
    Property UserId() As Int64
    Property WorkflowId() As Int64
    Property StepId() As Int64
    Property StepStateId() As Int64
    Property TaskStateId() As Int64
    Property SQL() As List(Of String)
    Property SQLCount() As List(Of String)
    Property SearchType As SearchTypes
    Property EntitiesEnabledForQuickSearch As List(Of IEntityEnabledForQuickSearch)
    Sub SetOrderBy(ByVal orderString As String)
    Sub SetGroupBy(ByVal orderString As String)
    Sub AddIndex(index As IIndex)
    Sub AddDocType(docType As IDocType)
    Property OrderBy As String

    Property Filters As List(Of ikendoFilter)
End Interface

File: Zamba/ZambaFramework/Zamba.FWBC/Zamba.FWBC/ISearch.cs
using System.Collections.Generic;
using Zamba;
using Zamba.Framework;
using Zamba.Searchs;

public interface ISearch
{
    List<IDocType> Doctypes   { get; set; }
List<IIndex>  Indexs   { get; set; }
   bool CaseSensitive  { get; set; }
    bool UseVersion  { get; set; }
    bool ShowIndexOnGrid  { get; set; }
    string Textsearch { get; set; }
    string ParentName  { get; set; }
    string Name { get; set; }
    int MaxResults  { get; set; }
    long UserId  { get; set; }
    long WorkflowId  { get; set; }
    long StepId  { get; set; }
    long StepStateId  { get; set; }
    long TaskStateId  { get; set; }
     List< string> SQL  { get; set; }
     List< string> SQLCount  { get; set; }
    SearchTypes SearchType { get; set; }
    List<IEntityEnabledForQuickSearch> EntitiesEnabledForQuickSearch  { get; set; }
    void SetOrderBy(string orderstring);
    void SetGroupBy(string orderstring);
    void AddIndex(IIndex index);
    void AddFilter(ikendoFilter filter);
    void AddDocType(IDocType docType);
    string OrderBy { get; set; }

    List<ikendoFilter> Filters { get; set; }
    string View { get; set; }
    string Restriction { get; set; }
    List<string> Lista_ColumnasFiltradas { get; set; }
}